File: app/src/main/java/com/example/dandelion/ui/ProfileFragment.java
package com.example.dandelion.ui;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.dandelion.EditProfileActivity;
import com.example.dandelion.R;
import com.example.dandelion.User;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.EventListener;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.FirebaseFirestoreException;

import de.hdodenhof.circleimageview.CircleImageView;

public class ProfileFragment extends Fragment {


    private FirebaseUser currentUser;
    private FirebaseFirestore mFirestore;

    private CircleImageView avatar;
    private TextView username;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.fragment_profile, container, false);

        currentUser = FirebaseAuth.getInstance().getCurrentUser();
        mFirestore = FirebaseFirestore.getInstance();
        username = root.findViewById(R.id.username);

        DocumentReference docRef = mFirestore.collection("users").document(currentUser.getUid());
        docRef.addSnapshotListener(new EventListener<DocumentSnapshot>() {
            @Override
            public void onEvent(@Nullable DocumentSnapshot value, @Nullable FirebaseFirestoreException error) {
                if (error != null) {
                    System.err.println("Listen failed: " + error);
                    return;
                }

                if (value != null && value.exists()) {
                    System.out.println("Current data: " + value.getData());
                    User user = value.toObject(User.class);
                    username.setText(user.getUsername());
                    //Glide.with(getApplicationContext()).load(user.getAvatarImageUrl()).into(mImageUri);

                } else {
                    System.out.print("Current data: null");
                }
            }
        });

        avatar = root.findViewById(R.id.avatar);
        avatar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getContext(), EditProfileActivity.class));
            }
        });
        return root;
    }
}

File: app/src/main/java/com/example/unlone/ui/profile/ProfileFragment.kt
package com.example.unlone.ui.profile

import com.google.firebase.auth.FirebaseUser
import com.google.firebase.firestore.FirebaseFirestore
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import com.google.firebase.auth.FirebaseAuth
import android.content.Intent
import android.view.View
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.unlone.databinding.FragmentProfileBinding
import com.example.unlone.instance.User
import com.google.firebase.firestore.EventListener
import com.google.firebase.firestore.ktx.toObject

class ProfileFragment : Fragment() {
    private var currentUser: FirebaseUser? = FirebaseAuth.getInstance().currentUser
    private var mFirestore: FirebaseFirestore? = FirebaseFirestore.getInstance()

    private var _binding: FragmentProfileBinding? = null
    private val binding get() = _binding!!

    private var profileList = mutableListOf<ProfileCard>()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentProfileBinding.inflate(inflater, container, false)
        val view = binding.root

        val recyclerView: RecyclerView = binding.recyclerviewProfile
        val profileAdapter = ProfileAdapter(requireActivity())
        val layoutManager = GridLayoutManager(activity, 2)
        recyclerView.layoutManager = layoutManager
        recyclerView.adapter = profileAdapter

        profileList.add(ProfileCard("My \nStories", "#EDC678"))
        // profileList.add(ProfileCard("Journey", "#E6E2CB"))   TODO
        profileList.add(ProfileCard("Saved", "#E2E0EE"))
        // profileList.add(ProfileCard("Setting", "#B0B0B0"))   TODO
        profileList.add(ProfileCard("Contact \nUs", "#B5CEF0"))
        profileList.add(ProfileCard("Log Out", "#B9CAB7"))
        profileAdapter.setDataList(profileList)

        val docRef = mFirestore!!.collection("users").document(
            currentUser!!.uid
        )
        docRef.addSnapshotListener(EventListener { value, error ->
            if (error != null) {
                System.err.println("Listen failed: $error")
                return@EventListener
            }
            if (value != null && value.exists()) {
                println("Current data: " + value.data)
                val user = value.toObject<User>()
                if (user != null) {
                    binding.name.text = user.username
                    binding.bio.text = user.bio
                }
            } else {
                print("Current data: null")
            }
        })

        binding.editButton.setOnClickListener{
            startActivity(
                Intent(
                    context,
                    EditProfileActivity::class.java
                )
            )
        }

        return view
    }
}